src/html/dom.pas

```diff
@@ -151,6 +155,10 @@       TElementChildrenNodeList = class (TNodeList)
         function GetItem(const Index: Integer): TNode; override;
         function GetFirstChild(): TNode; inline;
         function GetLastChild(): TNode; inline;
+        function GetFirstElementChild(): TElement inline;
+        function GetLastElementChild(): TElement inline;
+        property FirstElementChild: TElement read GetFirstElementChild;
+        property LastElementChild: TElement read GetLastElementChild;
```

Comment:
Why include these properties?

Reply:
I left it that way because of the calling code here:

https://github.com/whatwg/wattsi/blob/a9658ce5471c644374482f095ac5c886482075cb/src/html/dom.pas#L1615-L1619

And I wrote that code that way for parity with the existing `TElement.GetFirstChild()` code:

https://github.com/whatwg/wattsi/blob/a9658ce5471c644374482f095ac5c886482075cb/src/html/dom.pas#L1595-L1599

…which uses a property rather than function.

But if it seems better to avoid using the properties here at all for these new additions (since we’re not exposing them as public properties anyway), I can drop the (protected) properties for them and change the calling code to have `FChildNodes.GetFirstElementChild()`.

Lemme know if that’d be preferable.

Reply:
Yeah, I think not using properties would be better. In particular, the intention is to be intentionally inconsistent: getting the first child is just a lookup in a linked list, whereas getting the first element child is potentially expensive. So the former should use properties, but the latter should use methods.

Reply:
> Yeah, I think not using properties would be better. In particular, the intention is to be intentionally inconsistent: getting the first child is just a lookup in a linked list, whereas getting the first element child is potentially expensive.

OK, yup, makes sense — made it so